tools/hangar/__snapshots__/invalid.ts.snap

```diff
@@ -793,14 +793,14 @@ Duration <DURATION>"
 `;
 
 exports[`cloud_function_expects_inflight.test.w 1`] = `
-"error: Expected type to be \\"inflight (event: str): void\\", but got \\"preflight (name: str): str\\" instead
+"error: Expected phase to be preflight, but got inflight instead
```

Comment:
I feel a bit mixed on this new error because we're subtracting information away from the user. IMO we should try to find some way to highlighting the phase mismatch while still showing the expected and found types.

Two ideas that come to mind:

1) add on to the beginning of the error

```
error: Mismatched function phase; expected type to be "inflight (event: str): void", but got "preflight (name: str): str" instead
```

2) add the information as a separate error hint (see https://github.com/winglang/wing/issues/4503):

```
error: Expected type to be "inflight (event: str): void", but got "preflight (name: str): str" instead
  <code excerpt>
  hint: Expected phase to be inflight, but got preflight instead
```

What do you think about these alternatives?

(cc @MarkMcCulloh about error message style)

Reply:
I think this would work great as a hint to the existing type mismatch error 👍

Reply:
Hint sounds great.

Reply:
switched it to a hint like the others around it, now looking into the cli/lsp changes that need to be done for #4503

Reply:
added hints to diagnostics and tried to update the cli and lsp accordingly